staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -303,8 +308,10 @@ func (w *watchCache) waitUntilFreshAndBlock(resourceVersion uint64, trace *utilt
 	}
 	for w.resourceVersion < resourceVersion {
 		if w.clock.Since(startTime) >= blockTimeout {
-			// Timeout with retry after 1 second.
-			return errors.NewTimeoutError(fmt.Sprintf("Too large resource version: %v, current: %v", resourceVersion, w.resourceVersion), 1)
+			// Request that the client retry after 'resourceVersionTooHighRetrySeconds' seconds.
+			err := errors.NewTimeoutError(fmt.Sprintf("Too large resource version: %v, current: %v", resourceVersion, w.resourceVersion), resourceVersionTooHighRetrySeconds)
+			err.ErrStatus.Details.Causes = []metav1.StatusCause{{Message: "Too large resource version"}}
```

Comment:
nit: make "Too large resource version" a constant (it's already appearing 2 times here, and probably in other places too)

Reply:
You're right. This needs to be organized. I've moved the error creation and handling into storage/errors.go. I think this cleans things up nicely. Let me know what you think.